server/webhook.go

```diff
@@ -300,7 +299,7 @@ func (s *Server) processFile(ctx context.Context, pushEvent *vcs.PushEvent, repo
 		return "", false, nil, nil
 	}
 
-	schemaInfo, err := parseSchemaFileInfo(repo.BaseDirectory, repo.SchemaPathTemplate, file)
+	schemaInfo, err := db.ParseSchemaFileInfo(repo.BaseDirectory, repo.SchemaPathTemplate, file)
```

Comment:
We can enter Line 327 block if schemaInfo file is nil, is that expected?

I thought we only call ParseMigrationInfo if SchemaChangeType is DDL.

I am thinking the logic is:

if repo.Project.SchemaChangeType == api.ProjectSchemaChangeTypeDDL {
 db.ParseMigrationInfo
} else if repo.Project.SchemaChangeType == api.ProjectSchemaChangeTypeSDL {
 db.ParseSchemaFileInfo
}

Reply:
I need to change the processFile() logic entirely. We should allow DDL/DML(most importantly DML) even if SDL is enabled for the project.

The logic I'm thinking will be

```
schemaFileChange = db.ParseSchemaFileInfo()
migrationFileChange = db.ParseMigrationInfo()
if isSchemaFileChange {
  if repo.Project.SchemaChangeType == api.ProjectSchemaChangeTypeSDL {
    doSDLMigration()
  } else {
    // ignore because we don't allow SDL change for DDL project.
  }
}
```

Reply:
>> We should allow DDL/DML(most importantly DML) even if SDL is enabled for the project.

let's add this comment.

Also the comment in Line 322 ~ 323 is out-dated.

Reply:
Also Line 325 should be "Apply schema diff"

https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/plugin/db/driver.go?L321-335&subtree=true

Reply:
Done. Btw, the whole thing below needs to be refactored, so we don't need to make them right in this PR.